tests/test_arraycreation.py

```diff
@@ -117,6 +121,7 @@ def test_eye(N, M, k, dtype, order):
 @pytest.mark.usefixtures("allow_fall_back_on_numpy")
 @pytest.mark.parametrize("dtype", get_all_dtypes(no_float16=False))
 def test_frombuffer(dtype):
+    skip_dtype_not_supported(dtype)
```

Comment:
This is probably a bit late, but since you already has `get_all_dtypes` function you could consider and option to mark certain dtypes with `skip` flag inside it using `pytest.param`.

https://docs.pytest.org/en/7.3.x/how-to/parametrize.html

```python
# content of test_expectation.py
import pytest


@pytest.mark.parametrize(
    "test_input,expected",
    [("3+5", 8), ("2+4", 6), pytest.param("6*9", 42, marks=pytest.mark.xfail)],
)
def test_eval(test_input, expected):
    assert eval(test_input) == expected
```

If it is not already done, you probably also want API like this:

```python
@pytest.mark.parametrize("dtype", get_all_dtypes(['float', 'double', 'int32']))
```

I.e. make `get_all_dtypes` to accept list of types and mark unsupported with `skip` flag.